specification/integration_context.md

```diff
@@ -0,0 +1,91 @@
+<!-- markdownlint-configure-file 
+{ 
+  "MD013": { 
+    "tables": false 
+  } 
+}
+-->
+
+# Integration Context
+
+**Status**: [Experimental](../README.md#versioning-and-status-of-the-specification)
+
+In order to create a more complete picture of a distributed environment, we
+propose the use of additional headers to pass extra context details between
+Cisco observability components. Specifically, we are concerned with passing
+information between OpenTelemetry-based Splunk distributions and AppDynamics
+agents, in both directions.
+
+# Headers
+
+The following defines several new headers and describes their intended use.
+All headers MUST be treated as optional -- peer services will not always
+generate them. These headers SHOULD be treated as opaque values of type
+string.
+
+* `cisco-ctx-acct-id` - Contains the ID of the AppDynamics account.
+* `cisco-ctx-app-id` - Contains the ID of the AppDynamics application.
+* `cisco-ctx-tier-id` - Contains the ID of the AppDynamics tier.
+* `cisco-ctx-bt-id` - Contains the ID of the AppDynamics business transaction (BT).
+* `cisco-ctx-env` - Contains
+  the [deployment.environment.name](https://opentelemetry.io/docs/specs/semconv/attributes-registry/deployment/)
+  resource value from an OpenTelemetry based component.
+* `cisco-ctx-service` - Contains the [service.name](https://opentelemetry.io/docs/specs/semconv/resource/#service)
+  resource value from an OpenTelemetry based component.
+
+HTTP headers are capable of being multivalued. As such, implementations
+SHOULD use the _last_ value when the above headers contain multiple values.
+
+## Splunk OpenTelemetry distributions
```

Comment:
I think it would be helpful to describe how this can be implemented using OTel components (propagators, processors) based on https://github.com/signalfx/splunk-otel-java/pull/2198

Reply:
> Do I understand correctly that the propagator should be applied only in communication between Splunk and AppD?

How would one tell that context is being propagated to an app that is using appd? There is an open spec issue for controlling context propagation boundary. I hope that none of these headers are considered sensitive as limiting where they are propagated will be complicated.

Reply:
@laurit raises a good point that a customer might want the control of this behavior to be (optionally) more granular - e.g. `cisco.ctx.enabled`would be the preferred way, but also `cisco.ctx.enabled.inbound=true/false` and `cisco.ctx.enabled.outbound` as more specific toggles.  (If so, further spec issue to settle - what if the "combination" `enabled` is set and a more specific one is also set - which takes precedence?)

Reply:
For outbound injection, instrumentation is obviously not going to be able to tell what it's talking to.  For security/leakage concerns I've suggested more granular behavior controls, but the fallback could also be telling the user to configure a network proxy/firewall/etc. for their needs.  

This feature (appd<->splunk context sharing) needs to be off by default in `splunk-otel` per PM, who I'm sure would be  happy to share reasoning in separate channels.

Reply:
> Do I understand correctly that the propagator should be applied only in communication between Splunk and AppD?

No, this was not the intention. What phrasing leads you to believe this?

> How would one tell that context is being propagated to an app that is using appd?

You can't and shouldn't need to. The headers shouldn't contain sensitive information. Users who choose to put sensitive information in any of these fields (service name, environment, appd IDs, whatever) shouldn't use this feature.

Reply:
> What phrasing leads you to believe this?

This:

> See [integration_context.md](integration_context.md) for specifics about
exchanging additional context between AppD and splunk-otel based agents.

Threat 1:  Information disclosure
If I understand correctly the current design sends splunk-otel context to everyone (via outgoing requests).

Thread 2: Spoofing, Tampering?
The other problem is that the incoming requests are also not verifying who send the data (if this is actually coming from AppD). A malicious actor can send `cisco-ctx-acct-id` headers which would be stored in spans.

Reply:
While yes, technically, this approach of passing data in headers does disclose information, I don't think any of these fields can be considered sensitive. 

And I'm not sure what else to document -- this spec here is just describing what data goes in headers. If you have a service name or environment that is hyper secretive and you don't want it disclosed, sure, you shouldn't use this feature. However, I'm not sure that documenting it _here_ would help any end users of splunk-otel-* or appd agents. 

In hopes of moving this forward, I will add a note.